Update mentions of viam-provisioning.json to viam-defaults.json (#4110)
viamrobotics/docs: docs/manage/fleet/provision/setup.md

Reviewer: There are a lot of references to agent-provisioning which is no longer an actual thing. There is the Agent, which has a provisioning functionality, but no longer any seperate subsystem called "agent-provisioning."

Reviewer 2: Ok changing

If you are using the captive portal, this step is optional.
If you are using a mobile app, you must create a provisioning configuration file, specifying at least a `fragment_id`.

Create a file called <FILE>viam-defaults.json</FILE> with the following format and customize the [attributes](/manage/fleet/provision/setup/#configure-agent-provisioning):

{{< tabs >}}
{{% tab name="Template" %}}

```json {class="line-numbers linkable-line-numbers"}
{
  "network_configuration": {
    "manufacturer": "<NAME>", # your company name
    "model": "<NAME>", # the machine's model
    "fragment_id": "<ID>", # the fragment id, required for mobile app
    "hotspot_prefix": "<PREFIX>", # machine creates a hotspot during setup
    "disable_captive_portal_redirect": false, # set to true if using a mobile app
    "hotspot_password": "<PASSWORD>", # password for the hotspot

>     "networks" : []

Reviewer: The networks section isn't part of network_configuration.

```json {class="line-numbers linkable-line-numbers"}
{
  "network_configuration": {
    "manufacturer": "Skywalker",
    "model": "C-3PO",
    "fragment_id": "2567c87d-7aef-41bc-b82c-d363f9874663",
    "hotspot_prefix": "skywalker-setup",
    "disable_captive_portal_redirect": false,
    "hotspot_password": "skywalker123",
    "turn_on_hotspot_if_wifi_has_no_internet": false,
    "offline_before_starting_hotspot_minutes": "3m30s",
    "user_idle_minutes": "2m30s",

>     "retry_connection_timeout_minutes": "15m"

Reviewer: It's confusing that these are listed in the "example" but not the "template." Shouldn't the template be the exhaustive case? And example be the more minimal set with values actually filled out?

If you provide an app to your end user or are asking them to use the Viam mobile app, the user will provide network details through that app.

If you know in advance which other networks a machine should be able to connect to, we recommend that you add WiFi settings in the operating system (for example, directly in NetworkManager).

However, if you want to add additional networks to the provisioning configuration you can add them to the `networks` field value.

{{< alert title="Important" color="note" >}}
You must enable `turn_on_hotspot_if_wifi_has_no_internet` in the [`agent-provisioning` configuration](/manage/fleet/provision/setup/#configure-agent-provisioning) of the machine to allow the machine to connect to the specified networks after provisioning.

> {{< /alert >}}

Reviewer: It's not required, but it changes the criteria for a connection being "good enough." If the ONLY connection that works is an additional network, and it provides full internet, then it'll work without the setting. But if the "primary" network is available, (but doesn't provide internet) then it'll prefer that one, and consider it "good enough" even without internet. So it won't "try all networks until it gets internet" in that situation.

{
  "network_configuration": {
    "manufacturer": "Skywalker",
    "model": "C-3PO",
    "fragment_id": "2567c87d-7aef-41bc-b82c-d363f9874663",
    "hotspot_prefix": "skywalker-setup",
    "disable_captive_portal_redirect": false,
    "hotspot_password": "skywalker123",
    "turn_on_hotspot_if_wifi_has_no_internet": false,
    "offline_before_starting_hotspot_minutes": "3m30s",
    "user_idle_minutes": "2m30s",
    "retry_connection_timeout_minutes": "15m",
    "turn_on_hotspot_if_wifi_has_no_internet": true
  },

>   "additional_networks": [

Reviewer: Additional_networks is an object, not an array. Networks have to have a field name/label. I can't make the actual suggested change as the lines below aren't part of the diff, but it looks something like: 

```
    "additional_networks": {
      "testNet": {
        "priority": 30,
        "psk": "myFirstPassword",
        "ssid": "otherNetworkOne",
        "type": "wifi"
      }
    }
```

If you provide an app to your end user or are asking them to use the Viam mobile app, the user will provide network details through that app.

If you know in advance which other networks a machine should be able to connect to, we recommend that you add WiFi settings in the operating system (for example, directly in NetworkManager).

> If that is not possible, you can add networks throught the `additional_networks` field.

Reviewer 2: ```suggestion
If that is not possible, you can add networks with the `additional_networks` field.
```

```json {class="line-numbers linkable-line-numbers"}
{
  "network_configuration": {
    "manufacturer": "Skywalker",
    "model": "C-3PO",
    "fragment_id": "2567c87d-7aef-41bc-b82c-d363f9874663",
    "hotspot_prefix": "skywalker-setup",
    "disable_captive_portal_redirect": false,
    "hotspot_password": "skywalker123",
    "turn_on_hotspot_if_wifi_has_no_internet": false,
    "offline_before_starting_hotspot_minutes": "3m30s",
    "user_idle_minutes": "2m30s",
    "retry_connection_timeout_minutes": "15m",

>     "turn_on_hotspot_if_wifi_has_no_internet": true,

Reviewer: This is mentioned as being "required" for additional_networks to work in lots of text body above/around this. That's not the case. It changes behavior, but it's not required.

Also, that other text should (most likely) be removed. We should really discourage people from adding additional_networks in the defaults file.

>     "networks": [

Reviewer: There's no "networks" in network_configuration. There's a separate section for "additional_networks" and the format is now an object, not an array. Again, may be easier to just point to the main configuration page, as viam-defaults.json basically takes the exact same layout/format as the cloud config.

```

Follow the instructions.
If you created a <FILE>viam-defaults.json</FILE>, specify its location as an environment variable or when prompted.

You can set optional arguments by defining the following environment variables:

<!-- prettier-ignore -->
| Argument | Description |
| -------- | ----------- |
| `VIAM_JSON_PATH` | The path to the machine credentials <FILE>viam.json</FILE> file to be copied to the machine. The script will also prompt you for this file if not provided. |

> | `DEFAULTS_PATH ` | The path to the <FILE>viam-defaults.json</FILE> file. The script will also prompt you for this file if not provided. |

Reviewer: As I was worried about before, the version of the preinstall script with this change made wasn't yet uploaded. I've uploaded it now, so we should be good.

> | `DEFAULTS_PATH ` | The path to the <FILE>viam-defaults.json</FILE> file. The script will also prompt you for this file if not provided. |

Reviewer 2: ```suggestion
| `DEFAULTS_PATH` | The path to the <FILE>viam-defaults.json</FILE> file. The script will also prompt you for this file if not provided. |
```

> | `PROVISIONING_PATH` | The path to the <FILE>viam-defaults.json</FILE> file. The script will also prompt you for this file if not provided. |

Reviewer 2: ```suggestion
| `DEFAULTS_PATH ` | The path to the <FILE>viam-defaults.json</FILE> file. The script will also prompt you for this file if not provided. |
```

```

Follow the instructions.
If you created a <FILE>viam-defaults.json</FILE>, specify its location as an environment variable or when prompted.

You can set optional arguments by defining the following environment variables:

<!-- prettier-ignore -->
| Argument | Description |
| -------- | ----------- |
| `VIAM_JSON_PATH` | The path to the machine credentials <FILE>viam.json</FILE> file to be copied to the machine. The script will also prompt you for this file if not provided. |

> | `PROVISIONING_PATH` | The path to the <FILE>viam-defaults.json</FILE> file. The script will also prompt you for this file if not provided. |

Reviewer: This is called "DEFAULTS_PATH" now I believe, though I'm not sure if the changes to preinstall.sh with the new path were published publicly yet.

1. When you power on the machine that has `viam-agent` installed and `agent-provisioning` configured, `viam-agent` creates a WiFi hotspot.

>    - The [`agent-provisioning` configuration](/manage/fleet/provision/setup/#configure-agent-provisioning) is at <file>/etc/viam-defaults.json</file> on your machine.

Reviewer 2: ```suggestion
   - The [`defaults` file](/manage/fleet/provision/setup/#configure-agent-provisioning) is at <file>/etc/viam-defaults.json</file> on your machine.
```

1. When you power on the machine that has `viam-agent` installed and `agent-provisioning` configured, `viam-agent` creates a WiFi hotspot.

>    - The [`agent-provisioning` configuration](/manage/fleet/provision/setup/#configure-agent-provisioning) is at <file>/etc/viam-defaults.json</file> on your machine.

Reviewer: This isn't just the "agent-provisioning" config. As noted above, there is no longer anything specifically called "agent-provisioning." The defaults file is also not JUST for provisioning. It can be used to set ANY configuration value for agent to a preferred/custom default. Not sure if documentation explaining that belongs on this page or elsewhere though.

   Note that any features that require internet access will not function if the connected WiFi network is not connected to the internet.
   If you want `viam-agent` to require that a WiFi network be connected to the internet in order to connect to it, enable `turn_on_hotspot_if_wifi_has_no_internet`.

1. `viam-agent` then starts `viam-server` with the provided configuration and the machine becomes **live**.

{{% /tab %}}
{{% tab name="Captive web portal" %}}

1. When you, as the end user, power on the machine that has `viam-agent` installed, `viam-agent` creates a WiFi hotspot.

   The [`defaults` file](/manage/fleet/provision/setup/#configure-defaults) is at <file>/etc/viam-defaults.json</file>.
   If a machine already exists, a machine cloud credentials file, if provided, is at <file>/etc/viam.json</file>.

1. You as the end user then use your mobile device or computer and connect to the WiFi hotspot.

   By default, the hotspot network is named `viam-setup-HOSTNAME`, where `HOSTNAME` is replaced with the hostname of your machine.
   The WiFi password for this hotspot network is `viamsetup` by default.

>    You can customize these values in the [`defaults` file](/manage/reference/viam-agent/#defaults).

Reviewer 2: ```suggestion
   You can customize these values in the [`defaults` file](/manage/fleet/provision/setup/#configure-defaults).
```

1. When you, as the end user, power on the machine that has `viam-agent` installed and `agent-provisioning` configured, `agent-provisioning` creates a WiFi hotspot.

>    - The [`agent-provisioning` configuration](/manage/fleet/provision/setup/#configure-agent-provisioning) is at <file>/etc/viam-defaults.json</file>.

Reviewer 2: ```suggestion
   - The [`defaults` file](/manage/fleet/provision/setup/#configure-agent-provisioning) is at <file>/etc/viam-defaults.json</file>.
```
